pragma-node/src/handlers/entries/get_onchain/mod.rs

```diff
@@ -51,25 +50,37 @@ pub async fn get_onchain(
         now
     };
 
-    let (aggregated_price, sources) = get_sources_and_aggregate(
+    let (aggregated_price, sources, pair_used, price_decimals) = routing(
         &state.onchain_pool,
+        &state.offchain_pool,
         params.network,
         pair_id.clone(),
         timestamp,
         aggregation_mode,
+        is_routing,
     )
     .await
     .map_err(|db_error| db_error.to_entry_error(&pair_id))?;
 
     // TODO(akhercha): ⚠ gives different result than onchain oracle sometime
-    let last_updated_timestamp =
-        get_last_updated_timestamp(&state.onchain_pool, params.network, pair_id.clone())
+    let mut last_updated_timestamp = 0;
+    for pair in pair_used {
```

Comment:
Can we update `get_last_updated_timestamp` so instead of a `pair_id: String` it takes a `pair_ids: Vec<String>` argument?
So it works in one query for any number of pairs:
```sql
FROM
    {table_name}
WHERE
    pair_id in -- the list of pair ids...
ORDER BY timestamp DESC
LIMIT 1;
```

Reply:
done